### PR TITLE
Site Editor: Add Missing Styles

### PR DESCRIPTION
**Pull request by Copons**
## Description

- Add the `.block-editor` class to the Site Editor.
- Load experimental global styles in the Site Editor.

The Site Editor is initialized in a very different way compared to the Post Editor, and there are plenty of inconsistencies to be found.

I stumbled upon this issue when trying the HTML block in the Site Editor: its textarea didn't have any styles applied.
I then noticed that some styles depend on the `.block-editor` class, which is [created by Core WordPress](https://github.com/WordPress/WordPress/blob/0bc22bb1fc3a7ed3a9698cf72a1137c7443ad4ae/wp-admin/edit-form-blocks.php#L439) when rendering `edit-post`. `edit-site` is not initialized in the same way, and didn't have that class, making a few components and blocks (e.g. [`PlainText`](https://github.com/WordPress/gutenberg/blob/9db42e0462b7cf62d6db156cf211c379bd35f433/packages/block-editor/src/components/plain-text/style.scss#L1)) not working as intended.

After fixing the plain text view of the HTML block, I also noticed that its preview was broken as well.
As it turned out, we don't add the experimental global styles to the block-editor settings in `edit-site`.
I'm not sure why, though, and I can't tell if I'm causing any regressions here.
Pinging @nosolosw as author of the original PR for some pointers: https://github.com/WordPress/gutenberg/pull/24250

## How has this been tested?

- Open the Site Editor.
- Add the HTML block, and make sure it's styled appropriately.
- Insert some HTML, and click on "Preview" in the block toolbar.
- Make sure the preview is styled appropriately.

## Screenshots 

| Before | After |
| - | - |
| Screenshot 2020-11-13 at 12 34 40 | Screenshot 2020-11-13 at 12 32 44 |
| Screenshot 2020-11-13 at 12 34 46 | Screenshot 2020-11-13 at 12 32 53 |

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: lib/global-styles.php [hunk 1]
```diff
@@ -454,16 +454,15 @@ function_exists( 'gutenberg_is_edit_site_page' ) &&
 		$settings['__experimentalGlobalStylesUserEntityId'] = gutenberg_experimental_global_styles_get_user_cpt_id();
 		$settings['__experimentalGlobalStylesContexts']     = $base->get_blocks_metadata();
 		$settings['__experimentalGlobalStylesBaseStyles']   = $base->get_raw_data();
-	} else {
-		// STEP 3 - OTHERWISE, ADD STYLES
-		//
-		// If we are in a block editor context, but not in edit-site,
```
Comment (Addison-Stavlo) [inline lib/global-styles.php:460]: Interesting, I wonder why "not in edit-site" and if we are missing anything? 🤔

**Review by Addison-Stavlo — COMMENTED** [review]
Looks good to me!  Tested and it all now works as I would expect.

The only thing holding off an approval is the curiosity about "not in edit-site" comment, which I see you have ping'd @nosolosw for a response on that.

**Comment by nosolosw** [comment]
:wave: I've taken a look at this and want to share my thoughts. There are two issues here:

### Blocks that depend on the `.block-editor` class being present

I've searched for the `block-editor` selector and these blocks depend on it being present: archives, categories, latest-posts, more, rss, shortcode, tag-cloud. There are another set of blocks that depend on the selector through the dependency on the block-editor/components/plain-text: file, HTML, shortcode.

Most of these block's CSS hasn't been updated for 2 years, so it looks like a case where the new editors introduced didn't test that these blocks properly. Although the change here seems to work, I'm unsure if we should make all editor use the `.block-editor` selector or whether it's preferrable that we remove that selector from the blocks that have it? I'm largely unfamiliar with how these editors came to be and the role of this selector so pinging a few folks for context @noahtallen @vindl for FSE/edit-site, @youknowriad @jasmussen for post-editor, @draganescu @talldan for widgets screen. It looks like the navigation doesn't have this issue because none of the affected blocks are allowed here.

### Global styles in the site editor

These styles are present in the site editor, only that they're generated client-side (look for the embedded stylesheet with id `global-styles-inline-css`). The reason these are loaded differently for the site editor is that the user can update the styles through the Global Styles sidebar, so we need a way to consolidate the styles (core, theme, and user) upon user changes.

By looking at the issue description I'm unsure what's the issue? Would you mind expanding on this?

**Comment by Copons** [comment]
Thanks for looking at this @nosolosw!

> ### Global styles in the site editor
> These styles are present in the site editor, only that they're generated client-side (look for the embedded stylesheet with id `global-styles-inline-css`). The reason these are loaded differently for the site editor is that the user can update the styles through the Global Styles sidebar, so we need a way to consolidate the styles (core, theme, and user) upon user changes.
> 
> By looking at the issue description I'm unsure what's the issue? Would you mind expanding on this?

It's I think specific to the HTML block which loads the preview in an iframe.

That iframe needs the editor styles to properly display the content, and it grabs them from [`( 'core/block-editor' ).settings.styles`](https://github.com/WordPress/gutenberg/blob/6e4c2eb333ddc3cf3e87e4fa716cc18c3d3131fb/packages/block-library/src/html/edit.js#L38).
Since we are not registering the Global Styles settings in the Site Editor, the HTML preview iframe can't get them, and falls back to whatever default (see in the screenshots: before is incorrectly serif).

If I'm understanding the code correctly, on update Global Styles directly [modifies `<style id="global-styles-inline-css">`](https://github.com/WordPress/gutenberg/blob/6e4c2eb333ddc3cf3e87e4fa716cc18c3d3131fb/packages/edit-site/src/components/editor/global-styles-provider.js#L140) in the DOM.
I see that it's also supposed to [`updateSettings`](https://github.com/WordPress/gutenberg/blob/6e4c2eb333ddc3cf3e87e4fa716cc18c3d3131fb/packages/edit-site/src/components/editor/global-styles-provider.js#L148-L156), but the action diff in the Redux dev tools is empty. 🤔 
I guess we should make sure the inline style and the Redux settings are kept in sync.

**Comment by jasmussen** [comment]
Interestingly, I just ran into this and fixed it on my end: https://github.com/Automattic/block-experiments/pull/156

**Comment by nosolosw** [comment]
It seems that we have two different issues here:

1. The use of `.block-editor` selector in some blocks => we either have to add that class **to all editors** or remove the class from blocks.

https://github.com/WordPress/gutenberg/pull/27063 attempts to remove the class from the blocks that have it, as they shouldn't depend on a specific editor selector. I'd appreciate wide testing and feedback there.

2. There's some blocks that don't work well with how we generate the styles dynamically in edit-site.

HTML preview (uses iframe) has this issue. In a conversation with @jorgefilipecosta he brought up that other blocks may have issues as well (classic, shortcode, etc). I haven't confirmed but this needs investigation.

To fix this second issue we need to update how we generate the stylesheet client-side for edit-site: we need to be able to inject (and update upon user changes) our client-side generated stylesheet within the settings.styles. We also need to respect all the existing styles.

I'm going to start working on this (unless someone has already started).

**Comment by nosolosw** [comment]
PR for second issue at https://github.com/WordPress/gutenberg/pull/27065

**Comment by nosolosw** [comment]
Both PRs have been merged so this should be fixed. Thanks for reporting!

**Pull request closed without merging**